NebulaClient/PacketProcessors/Factory/Entity/BuildEntityRequestProcessor.cs

```diff
@@ -20,6 +20,7 @@ public void ProcessPacket(BuildEntityRequest packet, NebulaConnection conn)
                 using (FactoryManager.EventFromServer.On())
                 {
                     FactoryManager.EventFactory = planet.factory;
+                    FactoryManager.PacketAuthor = packet.AuthorId;
```

Comment:
Should these always be set? Maybe could find a way to save this context.

Reply:
This `AuthorId `is used to solve collisions such as: if two players build two buildings, the host will build the first one and denies the second one, and to the notification packet about the building being built, the host will attach the `AuthorId` so the correct player will remove the item from the inventory and add sand count.

The same is going for the building removal request when two players remove the same building at the same time, the host will include `authorId` for the removal request that was successful so the correct player will add item to the inventory.

Since adding and removing items from inventory + adding sand count is done in different files I have needed to add it here.

Reply:
Oh, I was thinking about having this exposed in the EventFromServer toggle, by creating an Author id scope or something.

Reply:
If you mean something like:
```cs
using (FactoryManager.EventFromServer.On(packet.AuthorId)) {

//usage:
FactoryManager.EventFromServer.AuthorId
```
I am not exactly sure how to modify the toggle to store extra information.

Reply:
I can modify it, it would mean that toggle could store a `Stack<TMetadata>` which would be pushed to in `On` and popped when disposed. Don't worry about it too much.